pingcap/tidb-operator
# delete orphan tidb info key to make dashboard happy

### What problem does this PR solve?


fix https://github.com/pingcap/tidb-operator/issues/3697

### What is changed and how does it work?


for every orphan  `/topology/tidb/{addr}/info` we delete it if the instance is not expected to be exist

for PD, TiKV, TiFlash, no need to do this, it will get the logical instances from pd

### Code changes


### Tests


- deploy a cluster with 3 tidb
- update replica to be 1 and check we can see only one in dashboard
- update tc to use a wrong image and delete the tidb pod(just to make tidb failed to start), and check tidb instance still exist at dashboard but are not connected


### Side effects


### Related changes


### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Delete tidb info key to make the instance disappear at dashboard after scale in
```
